clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/PollEvent.java

```diff
@@ -16,12 +16,12 @@
  */
 package org.apache.kafka.clients.consumer.internals.events;
 
-public class PollEvent extends ApplicationEvent {
+public class PollEvent extends CompletableApplicationEvent<Void> {
 
     private final long pollTimeMs;
 
     public PollEvent(final long pollTimeMs) {
-        super(Type.POLL);
+        super(Type.POLL, Long.MAX_VALUE);
```

Comment:
IMO, it's crucial to wait for `allConsumed` to be completed, so I set it to `Long.MAX_VALUE`.